### PR TITLE
added webworker version of console.html

### PR DESCRIPTION
**Pull request by joemarshall**
This adds another version of console.html which runs pyodide in a webworker. 

It can't be used for things that need to use js to talk to the window object, but for just python use, it is nice because you can e.g. run pytest tests and see the output in real time. 

No input or interrupt handling right now, although with jspi I guess it would be possible to do without too much hassle.

### CODE REVIEW

**Comment by ryanking13** [comment]
Thanks @joemarshall. It's nice to have an example of using webworker, but I'm not sure if we should include it in the final distribution.

I'd like to reduce the files included in the distribution to only what is necessary (#4942), so IMO this kind of file would be better added to the example section of the documentation (#4379, #4390).

**Comment by joemarshall** [comment]
Great - what's the state of examples? Are you going to merge #4390 as that seems like the obvious place for this. 

Is console.html going to move there also? Could maybe also add a `make examples` commandline that made examples as a subdirectory of dist (or in dist), so that anyone who wanted to test things could put all the examples into there automatically and ready to serve; it's nice how console.html etc. just turn up when you do a build, but I can see it doesn't need to be in a distribution by default any more.

I also have `run_pytest_in_pyodide.py` to contribute to examples also, which runs a full pytest test suite inside pyodide from command line (without requiring pytest collection to happen on the host machine like pytest-pyodide does). Used in arrow CI, and quite fiddly to write because of handling echoing of stdout / stderr correctly to host console in real-time requires a certain amount of fiddling in javascript + selenium (it's easy to make it output at the end of tests only, but then it just hangs while the pyodide call goes in, which is sub-optimal)

**Comment by ryanking13** [comment]
> Are you going to merge https://github.com/pyodide/pyodide/pull/4390 as that seems like the obvious place for this.

@hoodmane @agriyakhetarpal WDYT? Are you okay with having examples section in our docs? Note that there was some discussion about it: https://github.com/pyodide/pyodide/pull/4379.

**Comment by agriyakhetarpal** [comment]
Hi @ryanking13, thanks for the ping. I've gone through the previous discussions linked, and it's certainly useful to have an examples section in our own docs, +1 for merging that PR. There are some other examples with Pyodide that I know such as the one on https://learn.scientific-python.org/development/guides/repo-review/ by the Scientific Python cookiecutter template authors, which would be useful to link to. That said, having a collection of assorted/curated snippets that can perform regular tasks hosted on a separate website would be a great idea as well, similar to https://sphinx-extensions.readthedocs.io/en/latest/.

**Comment by ryanking13** [comment]
Alright, let's then merge this, so we can put examples in the Pyodide docs.

**Comment by ryanking13** [comment]
Alright, then, let's merge #4390 so that we can put examples like this in our docs.

**Comment by joemarshall** [comment]
@ryanking13 I've moved the console webworker to examples now. It uses a separate .js file, so I slightly modified the instructions for examples to say a single .html file, or a single html plus a same name javascript file, just because workers without separate JS files are always a bit of a hack.

**Comment by joemarshall** [comment]
Do examples need a changelog entry, or can we treat them as minor?

**Comment by hoodmane** [comment]
I don't think examples need changelog entries.

File: Makefile [hunk 1]
```diff
@@ -253,6 +253,14 @@ dist/console.html: src/templates/console.html
...
 	sed -i -e 's#{{ PYODIDE_BASE_URL }}#$(PYODIDE_BASE_URL)#g' $@
 
+dist/console_webworker.html: src/templates/console_webworker.html
+	cp $< $@
+	sed -i -e 's#{{ PYODIDE_BASE_URL }}#$(PYODIDE_BASE_URL)#g' $@
+
+dist/console_webworker.js: src/templates/console_webworker.js
+	cp $< $@
+	sed -i -e 's#{{ PYODIDE_BASE_URL }}#$(PYODIDE_BASE_URL)#g' $@
+
```
Comment (ryanking13) [inline Makefile:263]: This can be removed now.

**Review by ryanking13 — COMMENTED** [review]
Thanks @joemarshall!

**Pull request merged**